### ISSUE TITLE
Onboard maintainers to the release process

### ISSUE THREAD

**Issue by FFY00** [pypa/build#561]
Our release process requires that maintainers have their PGP key listed in https://pypa-build.readthedocs.io/en/stable/installation.html, and preferably have their key signed by one of the other maintainers.

@pypa/build-committers if you wish to get onboarded, please reply to the issue with your PGP key and preferred email (that is listed in the key).

I will send you an email with a simple file for you to sign, to verify you hold the key and have access to the key email. After that I will sign your key and push it to the keyserver. Afterwards, either me or you can open a PR adding the key to the install documentation page, I will add you as a PyPI maintainer when the PR lands.

**Comment by henryiii** [comment]
I can make a PR, or just put it here: https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2fdec9863e5e14c7bc429f27b9d0e45146a241e8 - any GH email should be fine there too (see https://keyserver.ubuntu.com/pks/lookup?search=0x2fdec9863e5e14c7bc429f27b9d0e45146a241e8&fingerprint=on&op=index for the emails)

**Comment by webknjaz** [comment]
Looks like this would help solve things like https://github.com/pypa/build/issues/615...

**Comment by henryiii** [comment]
We got stuck trying to get my PGP key signed. I think I need to export it then reimport it to pick up the new signatures. I'll try to do that soonish.

**Comment by webknjaz** [comment]
> I think I need to export it then reimport it to pick up the new signatures.

If you're talking about GitHub or GitLab, then yes — this is my experience in the past..

I'm wondering why people still don't go for sigstore these days...

**Comment by henryiii** [comment]
PGP support has been removed from PyPI:

https://blog.pypi.org/posts/2023-05-23-removing-pgp/

Personally, I'd be fine to move to trusted publisher releases.

**Comment by webknjaz** [comment]
> Personally, I'd be fine to move to trusted publisher releases.

I talked to Felipe during PyCon and he had strong feelings about signing stuff locally IIRC 🤷‍♂️

By the way, it's a one-liner, to add sigstore signing to GHA..

**Comment by webknjaz** [comment]
IIUC Felipe had strong feelings about signing locally.

FWIW adding sigstore signing to GHA is a one-liner..

**Comment by henryiii** [comment]
Yes, but it's simply no longer supported to sign stuff locally with PGP. It will be deleted on upload to PyPI.

**Comment by layday** [comment]
I don’t know if we’ve been uploading signatures to PyPI. As far as I’m aware Filipe signs git tags and that is it.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by henryiii** [comment]
The binaries were signed, I believe. How about this:

* Require signed git tags (with a list of allowed signatures) to build in CI.
* On a signed git tag, sigstore sign the binary on GHA.
* Publish with trusted publisher deployment.

Assuming you can check that a tag is signed & verify that the signature against a set of public keys, this should be secure enough? We just need to be as secure as Flit, Pip, and Installer. :)

**Comment by webknjaz** [comment]
> The binaries were signed, I believe.

I verified that no release of build has signatures attached on PyPI, meaning they were never uploaded.

**Comment by webknjaz** [comment]
> Publish with trusted publisher deployment.

Also, the job generating the dists (or better — a separate one) should run the sigstore action.

**Comment by webknjaz** [comment]
FWIW Filipe's reasoning was that since the GHA infra is a public platform with shared runners, it cannot be trusted at all because we can't rely on it isolating different jobs well enough.
So the attack vector he was thinking about is the runner compromise.

**Comment by henryiii** [comment]
Wouldn't requiring several maintainers to sign-off on the deployment environment help with that? Then GitHub itself would have to be compromised, I think, as the runner waits for GitHub?
